### PR TITLE
Flutter 2.10.0-0.1.pre

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

File: tool/pub_dependencies_beta.json [hunk 1]
```diff
@@ -1,53 +1,50 @@
...
   "async": "2.8.2",
-  "bloc": "8.0.1",
+  "bloc": "8.0.2",
   "boolean_selector": "2.1.0",
   "characters": "1.2.0",
   "charcode": "1.3.1",
   "clock": "1.1.0",
-  "cloud_firestore": "3.1.4",
-  "cloud_firestore_platform_interface": "5.4.9",
-  "cloud_firestore_web": "2.6.4",
```
Comment (RedBrogdon) [inline tool/pub_dependencies_beta.json:10]: Did they unfederate the plugin?

**Review by RedBrogdon — APPROVED** [review]

File: tool/pub_dependencies_beta.json (hunk 1 above)
Comment (domesticmouse) [reply tool/pub_dependencies_beta.json:10]: @puf might know what is going on here

Comment (domesticmouse) [reply tool/pub_dependencies_beta.json:10]: Actually, doing a bit of digging, `cloud_firestore` is a registerable firestore package: https://github.com/dart-lang/dart-services/blob/master/lib/src/project.dart#L67

@srawlins did we drop support for the registerable packages at some point?

**Comment by domesticmouse** [comment]
And it looks like Flutter 2.10 beta is breaking us: https://github.com/dart-lang/dart-services/runs/4797256993?check_suite_focus=true#step:9:793

File: tool/pub_dependencies_beta.json (hunk 1 above)
Comment (srawlins) [reply tool/pub_dependencies_beta.json:10]: No they should still be there. Maybe something changed with `flutter packages-get`...

**Review by srawlins — APPROVED** [review]

File: tool/pub_dependencies_beta.json (hunk 1 above)
Comment (domesticmouse) [reply tool/pub_dependencies_beta.json:10]: @srawlins can you have a go at replicating this PR? I can't see what I did wrong here.

Comment (srawlins) [reply tool/pub_dependencies_beta.json:10]: I think this is unrelated to flutter 2.10. I can reproduce the changes in this file without bumping flutter. I'll keep looking...

Comment (srawlins) [reply tool/pub_dependencies_beta.json:10]: I didn't see that the version numbers dropped dramatically. E.g. firebase_auth from `3.3.3` to `0.15.1+1`. 😱 . Curiouser and curiouser.

Comment (domesticmouse) [reply tool/pub_dependencies_beta.json:10]: With @srawlins PR, `cloud_firestore` is back

**Review by srawlins — APPROVED** [review]
All these version numbers look sane to me.

**Comment by srawlins** [comment]
If there is an issue with generated_plugin_registrant.dart, I'll have to look tomorrow.

**Comment by srawlins** [comment]
I can't reproduce this failure, either manually:

```none
$ ls project_templates/firebase_project/lib/
generated_plugin_registrant.dart
```

nor via `dart run test test/project_creator_test.dart`...

**Comment by srawlins** [comment]
Ah I can reproduce via `FLUTTER_CHANNEL=beta dart run test test/project_creator_test.dart`

**Comment by domesticmouse** [comment]
Hey @srawlins, I'm curious what the status is. This issue is a blocker for DartPad.

/cc @devoncarew

**Comment by srawlins** [comment]
I spent time on Friday debugging and couldn't figure out why the generated plugin registrant file wasn't being seen. It will take more work.

I would not block Flutter 2.10 on DartPad on those two test cases. As long as the Flutter and Firebase samples run, it's fine. We can skip them or something.

**Comment by devoncarew** [comment]
I haven't dug into this; is there an issue with the generated `generated_plugin_registrant.dart` file? We could check w/ somr Flutter engineers to see if there have been recent changes there.

**Comment by domesticmouse** [comment]
Okiday, landing as is and seeing if the resulting images compile and run

**Pull request merged**